cluster-support-bot.py

```diff
@@ -47,53 +40,62 @@ def dashboard_uri(cluster):
     return '{}{}'.format(dashboard_base, cluster)
 
 
-# Example responder to greetings
-@slack_events_adapter.on("app_mention")
-def handle_message(event_data):
+@slack.RTMClient.run_on(event='message')
+def handle_message(**payload):
     global recent_events
 
-    logger.debug('handle_message: {}'.format(event_data))
-    message = event_data['event']
-    if message.get('subtype') is not None:
-        return  # https://api.slack.com/events/message#message_subtypes
-    text = message.get('text')
-    if not text:
-        return
-
-    timestamp = float(message.get('ts', 0))
-    if timestamp in recent_events:  # high-resolution timestamps should have few false-negatives
-        logger.info('ignoring duplicate message: {}'.format(message))
-        return
-
-    recent_events.add(timestamp)  # add after check without a lock should be a small race window
-    cutoff = time.time() - 60*60  # keep events for an hour
-    recent_events = {timestamp for timestamp in recent_events if timestamp > cutoff}
-
-    user_arg_line, body = (text.strip()+'\n').split('\n', 1)
-    user_args = user_arg_line.split()[1:]  # split and drop the '<@{bot-id}>' prefix
     try:
-        args = parser.parse_args(user_args)
-    except HelpRequest as error:
-        handler = handle_help(slack_client=slack_client, event=event_data, subparser=error.args[0]['parser'])
-    except ValueError as error:
-        handler = handle_parse_args_error(slack_client=slack_client, event=event_data, error=error)
-    else:
-        handler = args.func
-        if not handler:
-            logger.info('no handler found for {!r}'.format(user_args))
+        data = payload.get('data', [])
+        if not data:
             return
-        response = handler(slack_client=slack_client, event=event_data, args=args, body=body)
-        if not response:
+        if data.get('subtype') is not None:
+            return  # https://api.slack.com/events/message#message_subtypes
+        text = data.get('text', [])
+        if not text:
+            return
+        bot_mention = "@" + bot_user
```

Comment:
No need to construct this when handling each message.  Just `bot_mention = '@' + os.environ['BOT_USER']` as a module global (where you currently set `bot_user`).

Reply:
done